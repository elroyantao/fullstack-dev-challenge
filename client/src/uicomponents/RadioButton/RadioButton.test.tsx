import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import RadioButtonGroup from './RadioButton'

const onChangeMock = jest.fn()

beforeEach(() => {
    onChangeMock.mockClear()
})

describe('<RadioButtonGroup />', () => {
    describe('@renders', () => {
        it('default state', () => {
            const wrapper = shallow(
                <RadioButtonGroup
                    options={['a', 'b']}
                    defaultValue={'a'}
                    name="name"
                    onChange={onChangeMock}
                />
            )
            expect(toJson(wrapper)).toMatchSnapshot()
            expect(onChangeMock).toHaveBeenCalledTimes(0)
        })
    })

    describe('@side-effect', () => {
        it('should called onchange with `a` when radio button with value="a" has been clicked', () => {
            const wrapper = shallow(
                <RadioButtonGroup
                    options={['a', 'b']}
                    defaultValue={'a'}
                    name="name"
                    onChange={onChangeMock}
                />
            )
            wrapper.find('RadioButton[value="a"]').simulate('change', 'a')
            expect(onChangeMock).toHaveBeenCalledTimes(1)
            expect(onChangeMock).toHaveBeenLastCalledWith('a')
        })
    })
})
