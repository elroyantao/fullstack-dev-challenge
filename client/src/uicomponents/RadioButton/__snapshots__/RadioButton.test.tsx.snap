// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButtonGroup /> @renders default state 1`] = `
<RadioButtonGroup
  defaultValue="a"
  name="name"
  onChange={[MockFunction]}
  options={
    Array [
      "a",
      "b",
    ]
  }
>
  <HStack
    role="radiogroup"
  >
    <Stack
      align="center"
      direction="row"
      role="radiogroup"
    >
      <Styled(div)
        __css={
          Object {
            "& > *:not(style) ~ *:not(style)": Object {
              "marginStart": "0.5rem",
              "marginTop": 0,
            },
          }
        }
        alignItems="center"
        className="chakra-stack"
        display="flex"
        flexDirection="row"
        role="radiogroup"
      >
        <div
          className="chakra-stack css-benynn"
          role="radiogroup"
        >
          <RadioButton
            enterKeyHint="a"
            isChecked={true}
            key=".$a"
            name="name"
            onChange={[Function]}
            value="a"
          >
            <Box
              as="label"
            >
              <label
                className="css-1u8qly9"
              >
                <input
                  checked={true}
                  name="name"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  style={
                    Object {
                      "border": "0px",
                      "clip": "rect(0px, 0px, 0px, 0px)",
                      "height": "1px",
                      "margin": "-1px",
                      "overflow": "hidden",
                      "padding": "0px",
                      "position": "absolute",
                      "whiteSpace": "nowrap",
                      "width": "1px",
                    }
                  }
                  type="radio"
                  value="a"
                />
                <Box
                  _checked={
                    Object {
                      "bg": "primary",
                      "borderColor": "primary",
                      "color": "white",
                      "fontWeight": "bold",
                    }
                  }
                  aria-hidden={true}
                  borderRadius="md"
                  borderWidth="1px"
                  cursor="pointer"
                  data-checked=""
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  px={5}
                  py={3}
                >
                  <div
                    aria-hidden={true}
                    className="css-1p0pe0n"
                    data-checked=""
                    onMouseDown={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                  >
                    a
                  </div>
                </Box>
              </label>
            </Box>
          </RadioButton>
          <RadioButton
            enterKeyHint="b"
            isChecked={false}
            key=".$b"
            name="name"
            onChange={[Function]}
            value="b"
          >
            <Box
              as="label"
            >
              <label
                className="css-1u8qly9"
              >
                <input
                  checked={false}
                  name="name"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  style={
                    Object {
                      "border": "0px",
                      "clip": "rect(0px, 0px, 0px, 0px)",
                      "height": "1px",
                      "margin": "-1px",
                      "overflow": "hidden",
                      "padding": "0px",
                      "position": "absolute",
                      "whiteSpace": "nowrap",
                      "width": "1px",
                    }
                  }
                  type="radio"
                  value="b"
                />
                <Box
                  _checked={
                    Object {
                      "bg": "primary",
                      "borderColor": "primary",
                      "color": "white",
                      "fontWeight": "bold",
                    }
                  }
                  aria-hidden={true}
                  borderRadius="md"
                  borderWidth="1px"
                  cursor="pointer"
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  px={5}
                  py={3}
                >
                  <div
                    aria-hidden={true}
                    className="css-1p0pe0n"
                    onMouseDown={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                  >
                    b
                  </div>
                </Box>
              </label>
            </Box>
          </RadioButton>
        </div>
      </Styled(div)>
    </Stack>
  </HStack>
</RadioButtonGroup>
`;
